### ISSUE TITLE
Block and Rescue not failing properly with an include:

### ISSUE THREAD

**Issue by gamuniz** [ansible/ansible#17882]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

Block and Rescue with includes
##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

Stock configuration file.
##### OS / ENVIRONMENT

cat os-release 
NAME="Ubuntu"
VERSION="14.04.5 LTS, Trusty Tahr"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 14.04.5 LTS"
VERSION_ID="14.04"
##### SUMMARY

When using an include: as the Rescue task, it fails but the playbook shows successful.
##### STEPS TO REPRODUCE

```

---
- name: testing blocks
  hosts: all
  remote_user: ec2-user
  become: yes

  tasks:
    - block:
       - name: copying in a block
         copy: src=/tmp/test1.txt dest=/tmp/loops
      rescue:
       - include: copyblock.yml
```

##### EXPECTED RESULTS

```
PLAY [testing blocks] **********************************************************

TASK [setup] *******************************************************************
ok: [52.37.5.169]

TASK [copying in a block] ******************************************************
fatal: [52.37.5.169]: FAILED! => {"failed": true, "msg": "the file_name '/tmp/test1.txt' does not exist, or is not readable"}

TASK [Rescuing using an include:] ******************************************************
fatal: [52.37.5.169]: FAILED! => {"failed": true, "msg": "the file_name '/tmp/test1.txt' does not exist, or is not readable"}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @/var/lib/awx/projects/_6__github/blocktest.retry

PLAY RECAP *********************************************************************
52.37.5.169                : ok=1    changed=0    unreachable=0    failed=1   
```
##### ACTUAL RESULTS

```
PLAYBOOK: blocktest.yml ********************************************************
1 plays in blocktest.yml

PLAY [testing blocks] **********************************************************

TASK [setup] *******************************************************************
<52.37.5.169> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<52.37.5.169> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/root/.ssh/gabemuniz.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 52.37.5.169 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439 `" && echo ansible-tmp-1475518957.56-271988875689439="` echo $HOME/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439 `" ) && sleep 0'"'"''
<52.37.5.169> PUT /tmp/tmpeptfEj TO /home/ec2-user/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439/setup
<52.37.5.169> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/root/.ssh/gabemuniz.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[52.37.5.169]'
<52.37.5.169> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<52.37.5.169> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/root/.ssh/gabemuniz.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r 52.37.5.169 '/bin/sh -c '"'"'chmod u+x /home/ec2-user/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439/ /home/ec2-user/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439/setup && sleep 0'"'"''
<52.37.5.169> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<52.37.5.169> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o 'IdentityFile="/root/.ssh/gabemuniz.pem"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ec2-user -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r -tt 52.37.5.169 '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-woczfnqonzswqpiynrhbexcafygcuzvt; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ec2-user/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439/setup; rm -rf "/home/ec2-user/.ansible/tmp/ansible-tmp-1475518957.56-271988875689439/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
ok: [52.37.5.169]

TASK [copying in a block] ******************************************************
task path: /var/lib/awx/projects/_6__github/blocktest.yml:9
fatal: [52.37.5.169]: FAILED! => {"failed": true, "msg": "the file_name '/tmp/test1.txt' does not exist, or is not readable"}

TASK [copying in a block] ******************************************************
task path: /var/lib/awx/projects/_6__github/copyblock.yml:3
fatal: [52.37.5.169]: FAILED! => {"failed": true, "msg": "the file_name '/tmp/test1.txt' does not exist, or is not readable"}

PLAY RECAP *********************************************************************
52.37.5.169                : ok=1    changed=0    unreachable=0    failed=0   

```

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next release (2.2 and also cherry-picked to `stable-2.1` if there is another 2.1.x release).

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by jimi-c** [comment]
Ignore the above, was supposed to reference #17894 in that commit.
